### ISSUE TITLE
rancher agent cannot connect to server after EC2 instance restart(changed instance type)

### ISSUE THREAD

**Issue by ndelitski** [rancher/rancher#2537]
In UI host is in 'reconnecting' state.
In rancher-agent logs i had
```
{"id":"3e294b3b-f123-4fa0-a2ed-4e515b5f9291","type":"error","links":{},"actions":{},"status":401,"code":"Unauthorized","message":"Unauthorized","detail":null}
gzip: stdin: not in gzip format
tar: Child returned status 1
tar: Error is not recoverable: exiting now
INFO: Starting agent for 5FCA99F0A26F0E3B5E63
INFO: Access Key: 5FCA99F0A26F0E3B5E63
INFO: Config URL: http://rancher.mydomain.com/v1
INFO: Storage URL: http://rancher.mydomain.com/v1
INFO: API URL: http://rancher.mydomain.com/v1
INFO: IP: 52.19.xxx.xxx
INFO: Port:
INFO: Required Image: rancher/agent:v0.8.2
INFO: Current Image: rancher/agent:v0.8.2
INFO: Using image rancher/agent:v0.8.2
INFO: Downloading agent http://rancher.mydomain.com/v1/configcontent/configscripts
{"id":"98401acc-7b84-4a01-9151-e80124fe9860","type":"error","links":{},"actions":{},"status":401,"code":"Unauthorized","message":"Unauthorized","detail":null}
gzip: stdin: not in gzip format
tar: Child returned status 1
tar: Error is not recoverable: exiting now
```
Then i killed all rancher agents running on this host, started new one. Rancher change host state in UI to active, but all containers renamed with guids. Redeploying services previously running on this host had no effect, now i have new containers and old with "guid" names...

**Comment by deniseschannon** [comment]
When you changed the instance type, did you get a new IP for the instance? That's why I could see it going into reconnecting state.

**Comment by ndelitski** [comment]
@deniseschannon yep, actually it is a normal situation that host change ip, but agent has died and failed to launch unless i clean ALL agents(main + network + others) from the host and create new rancher agent. Ok, would then all the deployed services work as before, but not... they also become invisible for rancher scheduler and it began to deploy new ones there. However when i have to connect by ssh to my hosts and clean rancher agents and other mess too - this is not what rancher supposed to do i think.

**Comment by ndelitski** [comment]
Reconnect is not a problem, it always reconnect sometimes, problem is what
agent has crushed totally and host became useless for Rancher, rancher
should handle this situations without my help

**Comment by deniseschannon** [comment]
Okay, I had actually forgotten what the behavior was of what happens in this use case. i finally was able to test it out. I started an AWS EC2 instance. Stopped the EC2 instance. Changed instance type and started the instance. 

Rancher went into a reconnecting state at first, but it eventually resolved and was an active host again. But the host IP address was actually incorrect. 

To recover the host, you just need to re-run the the agent container again (without removing any of the old rancher/agent containers) and you'd also need to pass in the `-e CATTLE_AGENT_IP=<new_IP>`. This allows the existing containers in the managed network to remain in the managed network (and not be have a "guid" names)
